### PR TITLE
code improvement for issue #60

### PR DESCRIPTION
**Pull request by Hayashi-Yudai**
Improve the code.

TODOs

- [x] Fix code to more pythonic style
    - Fix import style
    - Remove some unused imports 
- [x] Fix error in lint by flake8

Issue: https://github.com/dcmoura/spyql/issues/60

### CODE REVIEW

File: spyql/parser.py [hunk 1]
```diff
@@ -20,11 +22,8 @@
 
 
 def get_agg_funcs():
-    import spyql.agg
-    import inspect
```
Comment (Hayashi-Yudai) [inline spyql/parser.py:24]: PEP8 says "Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants." cf. https://peps.python.org/pep-0008/#imports

File: spyql/prof.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import cProfile
-import spyql.cli
```
Comment (Hayashi-Yudai) [inline spyql/prof.py:2]: Removed unused import

File: spyql/writer.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import sys
 import io
 
-from spyql.log import user_debug, user_error, user_error
```
Comment (Hayashi-Yudai) [inline spyql/writer.py:9]: Removed duplicated import

File: tests/main_test.py [hunk 4]
```diff
@@ -19,6 +15,8 @@
 from functools import reduce
 import sys
 
+logging.basicConfig(level=logging.INFO)
+
```
Comment (Hayashi-Yudai) [inline tests/main_test.py:19]: This should be after imports

File: tests/null_test.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
-from spyql.nulltype import *
-from spyql.sqlfuncs import *
```
Comment (Hayashi-Yudai) [inline tests/null_test.py:2]: wildcard imports are not recommended in PEP8.
cf. https://peps.python.org/pep-0008/#imports

**Comment by dcmoura** [comment]
@Hayashi-Yudai Please assign me as reviewer when you are done. You can also add Diogo and Chaves. Thank you!

**Comment by Hayashi-Yudai** [comment]
@dcmoura I want, but I cannot assign anyone. Is there something I should do?

image

**Comment by dcmoura** [comment]
Typically, you would only open the PR when it is ready to review. Since you opened a WIP I would say that you could leave a comment when you are done.

**Comment by Hayashi-Yudai** [comment]
I see, thank you! Please review my code

**Review by recharte — APPROVED** [review]
Hi @Hayashi-Yudai thank you for your contribution, these small improvements make a big difference. :clap:

File: spyql/cli.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
+import spyql
```
Comment (ricardocchaves) [inline spyql/cli.py:1]: ❓ I may be missing something super obvious, but why did you add this line? At a first glance doesn't look like it's needed.

File: spyql/parser.py [hunk 7]
```diff
@@ -1,9 +1,11 @@
 from spyql.quotes_handler import QuotesHandler
-import spyql.utils
-import spyql.log
+from spyql import utils
+from spyql import log
 from spyql.processor import Processor
 from spyql.writer import Writer
 import re
+from spyql import agg
```
Comment (ricardocchaves) [inline spyql/parser.py:7]: 🙃 I would place this line next to the other `spyql` imports, unless there is a specific reason (eg. PEP8 guideline) for it to be this way. Non blocking.

File: spyql/processor.py [hunk 8]
```diff
@@ -309,37 +303,39 @@ def eval_clause(self, clause, clause_exprs, mode="eval"):
...
                 f"could not evaluate {clause.upper()} clause",
                 main_exception,
                 vars=self.vars,
             )
 
     # main
-    def go(self, output_options, user_query_vars={}) -> QueryResult:
+    def go(
+        self, output_options, user_query_vars={}
+    ) -> Tuple[QueryResult, Dict[str, int]]:
```
Comment (ricardocchaves) [inline spyql/processor.py:331]: 💪

**Review by ricardocchaves — COMMENTED** [review]
Hi @Hayashi-Yudai! Thanks for taking the time to tackle this, good work 💪 Only have a couple of small comments.

File: spyql/cli.py (hunk 6 above)
Comment (Hayashi-Yudai) [reply spyql/cli.py:1]: I noticed at line 79,

```python
@click.version_option(version=spyql.__version__)
```

so I think this imports is needed.

**Review by ricardocchaves — APPROVED** [review]
LGTM 🚀

File: spyql/prof.py (hunk 2 above)
Comment (dcmoura) [reply spyql/prof.py:2]: This brakes the code if you try to run in "profiler" mode:

```
python3 -m spyql.prof "SELECT 1"
```

because of:

https://github.com/dcmoura/spyql/blob/8445350cef8c336978274fb9de4ad0d2359f2091/spyql/prof.py#L10

flake8 can't see the code being executed in the string.

So I suggest keeping the import and adding an inline ignore flag: 

```
import spyql.cli # noqa: F401
```


This also shows that we should add a test case that tests calling the profile, but we can do that in another PR.

File: tests/null_test.py (hunk 5 above)
Comment (dcmoura) [reply tests/null_test.py:2]: OK. I must confess that I don't like the idea of maintaining the `spyql.sqlfuncs` list of functions, which is expected to grow. But I guess that's not a problem right now :-)

**Review by dcmoura — CHANGES_REQUESTED** [review]
Well done! 👏 

There was one change that breaks the code when running on profiler mode, but apart from that all looks better now! The tests currently do not test calling the profiler, so this was not easy to catch. 

The comments were super useful, thank you for including them. 

I did some performance checks and they were all good.
I also run `make lint` and the result was empty 🎉 😄

File: spyql/prof.py (hunk 2 above)
Comment (Hayashi-Yudai) [reply spyql/prof.py:2]: Oh... I fixed it.

Comment (dcmoura) [reply spyql/prof.py:2]: OK, now we are good :-)

**Review by dcmoura — APPROVED** [review]
👏 Thank you @Hayashi-Yudai! This was great! 

It was a pleasure having you contributing to spyql. 
We would be happy to see more contributions from you 😄 
Let me know if there's anything in the list of issues that you would like to tackle, or if there is some new feature that you would like to propose.
